automation/smoke-test-firefox-ios.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+
+# Purpose: Run Firefox-iOS tests against this application-services working tree.
+# Usage: ./automation/smoke-test-firefox-ios.py
+
+import argparse
+import os
+import subprocess
+import tempfile
+from shared import step_msg, fatal_err, run_cmd_checked
+
+parser = argparse.ArgumentParser(description="Run Firefox-iOS tests against this application-services working tree. "
+                                             "This script *MUST* be run from the root of the A-S repository.")
+group = parser.add_mutually_exclusive_group()
+group.add_argument("--use-local-repo",
+                    metavar="LOCAL_REPO_PATH",
+                    help="Use a local copy of firefox-ios instead of cloning it.")
+group.add_argument("--remote-repo-url",
+                    metavar="REMOTE_REPO_PATH",
+                    help="Clone a different firefox-ios repository.")
+parser.add_argument("--branch",
+                    default="master",
+                    help="Branch of firefox-ios to use.")
+parser.add_argument("--action",
+                    choices=["open-project", "run-tests", "do-nothing"],
+                    help="Run the following action once firefox-ios is set up.")
+
+DEFAULT_REMOTE_REPO_URL="https://github.com/mozilla-mobile/firefox-ios.git"
+
+args = parser.parse_args()
+firefox_ios_branch = args.branch
+local_repo_path = args.use_local_repo
+remote_repo_url = args.remote_repo_url
+action = args.action
+
+repo_path = local_repo_path
+
+if local_repo_path is None:
+    repo_path = tempfile.mkdtemp(suffix="-fxios")
```

Comment:
Once disadvantage of this is that by default, you have to do a full download and rebuild every time you run the script. I wonder if it's worth trying to discover a sensible default local repo, such as looking in `../firefox-ios` or similar. Or does that seem too risky in the case that's *not* what the user intended?

Reply:
Yeah, I'm very wary of messing up with somebody's worktree without their explicit consent, they can still use the `--use-local-repo` to use the same instance of firefox-ios every time.